src/commands/BulkCommand.ts

```diff
@@ -0,0 +1,82 @@
+import { Message, User } from 'discord.js';
+import PrefixCommand from './PrefixCommand';
+import MentionCommand from './MentionCommand';
+import ResolveRequestMessageTask from '../tasks/ResolveRequestMessageTask';
+import TaskScheduler from '../tasks/TaskScheduler';
+import emojiRegex = require( 'emoji-regex/text.js' );
+import { RequestsUtil } from '../util/RequestsUtil';
+import BotConfig from '../BotConfig';
+
+export default class BulkCommand extends PrefixCommand {
+	public readonly aliases = ['bulk', 'filter'];
+
+	public static currentBulkReactions: Map<User, Message[]>;
+
+	public async run( message: Message, args: string ): Promise<boolean> {
+		let rawEmoji: string;
+
+		if ( args.length ) {
+			const customEmoji = /^<a?:(\w+):(\d+)>/;
+			const unicodeEmoji = emojiRegex();
+
+			if ( customEmoji.test( args ) || unicodeEmoji.test( args ) ) {
+				rawEmoji = args;
+				const emojiMatch = customEmoji.exec( args );
+				if ( emojiMatch ) {
+					rawEmoji = emojiMatch[2];
+				}
+			} else {
+				await message.channel.send( `**Error:** ${ args } is not a valid emoji.` );
+				return false;
+			}
+		}
+
+		let ticketKeys: string[];
+		let firstMentioned: string;
+
+		try {
+			const bulkMessages = BulkCommand.currentBulkReactions.get( message.author );
+			let originMessages: Message[];
+			for ( const bulk of bulkMessages ) {
+				originMessages.push( await RequestsUtil.getOriginMessage( bulk ) );
+				await bulk.reactions.cache.get( BotConfig.request.bulkEmoji ).users.remove( message.author );
+			}
+			originMessages.forEach( origin => this.getTickets( origin.content ).forEach( ticket => ticketKeys.push( ticket ) ) );
```

Comment:
The nested `forEach` here is confusing.
```suggestion
			originMessages.forEach( origin => ticketKeys.push( ...this.getTickets( origin.content ) ) );
```